dlib/python/numpy_image.h

```diff
@@ -70,11 +73,9 @@ class numpy_rgb_image
     numpy_rgb_image (boost::python::object& img) 
     {
         long shape[3];
-        get_numpy_ndarray_parts(img, _data, shape);
+        get_numpy_ndarray_parts(img, _data, _contig_buf, shape);
         _nr = shape[0];
         _nc = shape[1];
-        if (shape[2] != 3)
```

Comment:
Nope. It's checked in `get_numpy_ndarray_parts` to ensure successful data copying. Anyway, do you have any suggestion on how to reduce the complexity? (i.e. to reuse as more code as possible for both checking and constructing)